src/base.strategy.ts

```diff
@@ -0,0 +1,46 @@
+import { RestInput } from "./types"
+import { Octokit } from "@octokit/core"
+import { throttling } from "@octokit/plugin-throttling"
+import { getOctokit } from "@actions/github"
+import { GitHub } from "@actions/github/lib/utils"
+import { RequestOptions } from "@octokit/types"
+
+export abstract class BaseStrategy {
+  protected setupClient(input: RestInput): InstanceType<typeof GitHub> {
+    const customClient = Octokit.plugin(throttling)
+    let customOctokit
+    if (!input.rateLimit) {
+      customOctokit = new customClient({ auth: input.token, throttle: { enabled: false } })
+    } else {
+      customOctokit = new customClient({
+        auth: input.token,
+        throttle: {
+          enabled: true,
+          onRateLimit: (retryAfter: number, options: RequestOptions) => {
+            console.warn(`Request quota exhausted for request ${options.method} ${options.url}`)
```

Comment:
In GitHub Actions, logging should be done through the dedicated functions from `@actions/core`. See `action.ts` for examples.

Reply:
removed console logging in favour of the github logging functions